### PR TITLE
Fix drop for KeeperMap

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

cc @tavplubix 

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Review by tavplubix — APPROVED** [review]

File: src/Storages/StorageKeeperMap.cpp [hunk 1]
```diff
@@ -494,9 +494,18 @@ void StorageKeeperMap::drop()
     checkTable<true>();
     auto client = getClient();
 
-    client->remove(table_path);
+    // we allow ZNONODE in case we got hardware error on previous drop
+    if (auto code = client->tryRemove(table_path); code == Coordination::Error::ZNOTEMPTY)
+    {
+        throw zkutil::KeeperException(
+            code, "{} contains children which shouldn't happen. Please DETACH the table if you want to delete it", table_path);
```
Comment (antonio2368) [inline src/Storages/StorageKeeperMap.cpp:501]: Maybe have this as logical error?

File: src/Storages/StorageKeeperMap.cpp [hunk 2]
```diff
@@ -494,9 +494,18 @@ void StorageKeeperMap::drop()
...
+    }
 
-    if (!client->getChildren(tables_path).empty())
+    std::vector<std::string> children;
+    // if the the tables is not found, some other table removed it
```
Comment (antonio2368) [inline src/Storages/StorageKeeperMap.cpp:505]:
```suggestion
    // if the tables_path is not found, some other table removed it
```

File: src/Storages/StorageKeeperMap.cpp (hunk 1 above)
Comment (tavplubix) [reply src/Storages/StorageKeeperMap.cpp:501]: Not sure, maybe we will add children in future versions

**Comment by tavplubix** [comment]
Stress test (debug) - server was not stopped gracefully within timeout because it was waiting for `MutateTask` to finish:
```
2022.09.22 13:58:12.668458 [ 1035822 ] {} <Information> Application: Received termination signal (Terminated)
...
2022.09.22 13:58:24.777001 [ 1035918 ] {c6834d43-6aea-46b7-896f-937c18eb3ffa::all_1_1_0_19992} <Trace> InterpreterSelectQuery: FetchColumns -> Complete
2022.09.22 13:58:31.666705 [ 1035833 ] {} <Trace> KeeperTCPHandler: Received heartbeat for session #1
2022.09.22 13:58:38.783341 [ 1035918 ] {c6834d43-6aea-46b7-896f-937c18eb3ffa::all_1_1_0_19992} <Debug> test_23.many_mutations (c6834d43-6aea-46b7-896f-937c18eb3ffa) (SelectExecutor): Key condition: unknown
2022.09.22 13:58:38.783582 [ 1035918 ] {c6834d43-6aea-46b7-896f-937c18eb3ffa::all_1_1_0_19992} <Debug> test_23.many_mutations (c6834d43-6aea-46b7-896f-937c18eb3ffa) (SelectExecutor): Selected 1/1 parts by partition key, 1 parts by primary key, 1/1 marks by primary key, 1 marks to read from 1 ranges
2022.09.22 13:58:38.783831 [ 1035918 ] {c6834d43-6aea-46b7-896f-937c18eb3ffa::all_1_1_0_19992} <Trace> MergeTreeInOrderSelectProcessor: Reading 1 ranges in order from part all_1_1_0, approx. 2 rows starting from 0
2022.09.22 13:58:41.666704 [ 1035833 ] {} <Trace> KeeperTCPHandler: Received heartbeat for session #1
2022.09.22 13:58:51.666701 [ 1035833 ] {} <Trace> KeeperTCPHandler: Received heartbeat for session #1
2022.09.22 13:59:01.666702 [ 1035833 ] {} <Trace> KeeperTCPHandler: Received heartbeat for session #1
2022.09.22 13:59:11.666710 [ 1035833 ] {} <Trace> KeeperTCPHandler: Received heartbeat for session #1
2022.09.22 13:59:12.273439 [ 1035918 ] {} <Debug> MemoryTracker: Peak memory usage to apply mutate/merge in c6834d43-6aea-46b7-896f-937c18eb3ffa::all_1_1_0_19992: 604.56 MiB.
...
2022.09.22 13:59:41.882560 [ 1035818 ] {} <Fatal> Application: Child process was terminated by signal 9 (KILL). If it is not done by 'forcestop' command or manually, the possible cause is OOM Killer (see 'dmesg' and look at the '/var/log/kern.log' for the details).
```

It's because `merges_blocker` was checked in `prepare()` only, but it should be checked before each step - https://github.com/ClickHouse/ClickHouse/pull/41699

**Pull request merged**